server/szurubooru/search/executor.py

```diff
@@ -94,7 +94,7 @@ def execute(
                 disable_eager_loads = True
 
         key = (id(self.config), hash(search_query), offset, limit)
-        if cache.has(key):
+        if not disable_eager_loads and cache.has(key):
```

Comment:
This change is good but the code is not very intuitive. Please add a note for me to refactor this in form of `  # TODO: refactor this` before the condition check.

Reply:
But there's still the issue that this breaks bulk editors. Maybe I should change my approach, and be more explicit about saving post order in client session?
Or I could modify the client again, to not bypass cache when doing bulk edits.

Reply:
I just realized that this also breaks pagination. The request for the next page is randomized, so it can contain some posts from the previous pages.

Reply:
Yes, I talked about it in some other issues. TBH I don't care about this anymore.

Reply:
But surely you wouldn't want to merge code that introduces blatant bugs like this?

Anyway, I think I figured it out: I'll store `cacheNumber` in context, and only update it when the user clicks the "randomize" button. That should solve all the above issues. I will overwrite this PR.

Reply:
Yes, by merging I meant after the bugfixes, of course. I like the idea of keeping the cache number.

Reply:
Wait, it looks like pagination with `sort:random` is already broken at `master`: some posts appear again on subsequent pages.

Reply:
Yes, the post do reappear due to the way the `ORDER BY` clause is formed, but the pagination remains consistent due to backend caching.